aten/src/ATen/UndefinedType.cpp

```diff
@@ -3,7 +3,7 @@
 namespace at {
 
 UndefinedType::UndefinedType(Context* context)
-: Type(context) {}
+: Type(context, /*is_variable=*/true) {}
```

Comment:
This seems like it can cause problems down the road.  For instance, in the JIT we might want to assert that something is _not_ a variable, but then we would need remember to write:

`something.defined() && !something.variable()`. 

Having it be `false` is bad as well. Since this is an internal check, maybe we should be more precise: `is_variable_or_undefined()`